cpp_exercise/src/lib/ipcmsgqlib.cc

```diff
@@ -0,0 +1,166 @@
+#include "ipcmsgqlib.h"
+#include "ipcexceptionlib.h"
+
+
+/***
+ * 
+ * MsgQueueSender Class
+ * 
+ * 
+ * **/
+
+MsgQueueSender::MsgQueueSender(const std::string & filename): _file_name(filename){
+        std::cout<<"**************************************************************"<<std::endl;
+        std::cout<<"****************************IPCSender: MESSAGE QUEUE PROTOCOL**************************"<<std::endl;
+        std::cout<<"**************************************************************"<<std::endl;
+        std::cout<<"**************************************************************"<<std::endl;
+
+   
+
+    int err;
+    remove(_msg_queue_name.c_str());
+        ret = mq_unlink(_msg_queue_name.c_str());
+
+    
+     memset(&_attrs, 0, sizeof _attrs);
```

Comment:
Why are you doing this? 
``` 
The fields of the struct mq_attr pointed to attr specify the
       maximum number of messages and the maximum size of messages that
       the queue will allow.  This structure is defined as follows:

           struct mq_attr {
               long mq_flags;       /* Flags (ignored for mq_open()) */
               long mq_maxmsg;      /* Max. # of messages on queue */
               long mq_msgsize;     /* Max. message size (bytes) */
               long mq_curmsgs;     /* # of messages currently in queue
                                       (ignored for mq_open()) */
           };

       Only the mq_maxmsg and mq_msgsize fields are employed when
       calling mq_open(); the values in the remaining fields are
       ignored.
``` 
From https://man7.org/linux/man-pages/man3/mq_open.3.html

Reply:
FIxed